src/units/abilities.cpp

```diff
@@ -1338,23 +1333,61 @@ static bool overwrite_special_affects(const config& special)
 	return (apply_to == "one_side" || apply_to == "both_sides");
 }
 
-unit_ability_list attack_type::overwrite_special_checking(unit_ability_list input, unit_ability_list overwriters, bool is_special) const
+unit_ability_list attack_type::overwrite_special_overwriter(unit_ability_list overwriters, const std::string& tag_name) const
 {
-	for(unit_ability_list::iterator i = overwriters.begin(); i != overwriters.end();) {
-		if(!overwrite_special_affects(*i->ability_cfg)) {
-			i = overwriters.erase(i);
-		} else {
-			++i;
+	std::vector<double> priorityvec;
+	int list_count = 0;
```

Comment:
This ends up being the number of elements in `overwriters`, correct?

Reply:
list_count count number of element in list because class ability_list don't have function for enumerate number of element.

Reply:
Please instead add a `size()` method to `unit_ability_list`, which returns the size of `cfgs_`.

Reply:
done